force-app/main/default/classes/ListDonation.cls

```diff
@@ -0,0 +1,215 @@
+/*
```

Comment:
I'm unsure about this class name. It's a generic name, starting with "list", but the implementation is specific to this component. It also has a lot of `public static` methods. Are all of these methods called externally? Is there a reason it's all static versus an instance service class called by a controller for the LWC (one that would have the @AuraEnabled methods). Basically, I think the architecture for the class structure should be reviewed and standarized. @bdvorachek @psadie-sfdo What do you think?

Also, as a new class, the copyright should be 2021, and as noted earlier all new classes, components, etc. should be at api v53.

Reply:
A better name here would be `DonationHistoryController`. Any class that has @AuraEnabled methods is a controller, and ideally it only has @AuraEnabled methods.

Inside of any @AuraEnabled static method in DonationHistoryController, we would spin up an instance of DonationHistoryService, probably providing the Id of the Contact as a constructor argument. All of the non-@AuraEnabled methods would get converted from static to being instance methods inside of DonationHistoryService. Example of how getDonationHistory would look in this controller class:

```
public static List<DonationHistoryView> getDonationHistory(Id contactId) {
    DonationHistoryService service = new DonationHistoryService(contactId);
    try {
        return service.getDonationHistory();
    } catch(Exception ex) {
        throw new AuraHandledException(ex.getMessage());
    }
}
```